pennylane/qaoa/cost.py

```diff
@@ -244,3 +244,214 @@ def maxcut(graph):
     )
     H = edge_driver(graph, ["10", "01"]) + identity_h
     return (H, qaoa.x_mixer(graph.nodes))
+
+
+def max_independent_set(graph, constrained=True):
+    r"""Returns the QAOA cost Hamiltonian and the recommended mixer corresponding to the MaxIndependentSet problem,
+    for a given graph.
+
+    Given some graph :math:`G`, an independent set is a set of vertices such that no two of the vertices in the set
+    share a common edge. The goal of MaxIndependentSet is to find the largest such set.
+
+    Args:
+        graph (nx.Graph): a graph defining the pairs of wires on which each term of the Hamiltonian acts
+        constrained (bool): specifies the variant of QAOA that is performed (constrained or unconstrained)
+
+    Returns:
+        (.Hamiltonian, .Hamiltonian):
```

Comment:
We are returning two Hamiltonians? :confused:

Reply:
Ah maybe this is a case where adding a sentence saying that it's the cost and mixer hamiltonians together, would be useful